classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -35,163 +34,249 @@ protected function getContextSettingsTable(): string
+        $this->addTableProcessor('subscription_type_settings', function (): int {
+            $affectedRows = 0;
+            // Depends directly on ~1 entities: type_id->subscription_types.type_id
+            // Custom field (not found in at least one of the softwares)
+            $affectedRows += $this->deleteRequiredReference('subscription_type_settings', 'type_id', 'subscription_types', 'type_id');
+            return $affectedRows;
+        });
 
-            // Clean orphaned completed_payments entries by context_id
-            $orphanedIds = DB::table('completed_payments AS cp')->leftJoin('journals AS j', 'j.journal_id', '=', 'cp.context_id')->whereNull('j.journal_id')->distinct()->pluck('cp.context_id');
-            foreach ($orphanedIds as $journalId) {
-                DB::table('completed_payments')->where('context_id', '=', $journalId)->delete();
+        // Support for the issueId setting
+        $this->addTableProcessor('publication_settings', function (): int {
+            $affectedRows = 0;
+            $rows = DB::table('publications AS p')
+                ->join('publication_settings AS ps', 'ps.publication_id', '=', 'p.publication_id')
+                ->leftJoin('issues AS i', DB::raw('CAST(i.issue_id AS CHAR(20))'), '=', 'ps.setting_value')
+                ->where('ps.setting_name', 'issueId')
+                ->whereNull('i.issue_id')
+                ->get(['p.submission_id', 'p.publication_id', 'ps.setting_value']);
+            foreach ($rows as $row) {
+                $this->_installer->log("The publication ID ({$row->publication_id}) for the submission ID {$row->submission_id} is assigned to an invalid issue ID \"{$row->setting_value}\", its value will be updated to NULL");
+                $affectedRows += DB::table('publication_settings')
+                    ->where('publication_id', '=', $row->publication_id)
+                    ->where('setting_name', 'issueId')
+                    ->where('setting_value', $row->setting_value)
+                    ->delete();
             }
+            return $affectedRows;
+        });
+    }
 
-            // Clean orphaned completed_payments entries by user_id
-            $orphanedIds = DB::table('completed_payments AS cp')->leftJoin('users AS u', 'u.user_id', '=', 'cp.user_id')->whereNull('u.user_id')->distinct()->pluck('cp.user_id');
-            foreach ($orphanedIds as $userId) {
-                DB::table('completed_payments')->where('user_id', '=', $userId)->delete();
-            }
-        } catch (\Exception $e) {
-            if ($fallbackVersion = $this->setFallbackVersion()) {
-                $this->_installer->log("A pre-flight check failed. The software was successfully upgraded to {$fallbackVersion} but could not be upgraded further (to " . $this->_installer->newVersion->getVersionString() . '). Check and correct the error, then try again.');
-            }
-            throw $e;
-        }
+    protected function getEntityRelationships(): array
```

Comment:
This list is specific for each application, I'll write a comment in the issue with the queries that I used to generate the cleanup code and this list.